elements/lisk-transaction-pool/src/transaction_pool.ts

```diff
@@ -413,35 +384,57 @@ export class TransactionPool {
 			}
 			const processableTransactions = txList.getProcessable();
 			const allTransactions = [...processableTransactions, ...promotableTransactions];
-			let firstInvalidTransaction: { id: Buffer; status: TransactionStatus } | undefined;
-			try {
-				await this._applyFunction(allTransactions);
-			} catch (error) {
-				const failedStatus = this._getStatus(error as TransactionFailedResponse);
-				firstInvalidTransaction = {
-					id: (error as TransactionFailedResponse).id,
-					status: failedStatus,
-				};
+			const veirfyResults = [];
+			for (const transaction of allTransactions) {
+				try {
+					const status = await this._verifyFunction(transaction);
+					veirfyResults.push({
+						id: transaction.id,
+						status,
+						transaction,
+					});
+					if (status === TransactionStatus.INVALID) {
+						break;
+					}
+				} catch (error) {
+					veirfyResults.push({
+						id: transaction.id,
+						status: TransactionStatus.INVALID,
+						transaction,
+					});
+					break;
+				}
```

Comment:
It seems to me that this loop and the very next one `for (const tx of allTransactions)` could be combined in one?